test/src/experimental_helpers.cc

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file experimental_helpers.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+#include "experimental_helpers.h"
+
+namespace tiledb::test {
+
+void add_dimension_label(
+    tiledb_ctx_t* ctx,
+    tiledb_array_schema_t* array_schema,
+    const std::string& label_name,
+    const uint32_t dim_idx,
+    tiledb_label_order_t label_order,
+    tiledb_datatype_t label_datatype,
+    const void* label_domain,
+    const void* label_tile_extent,
+    const void* index_tile_extent,
+    optional<uint32_t> label_cell_val_num,
```

Comment:
I like using optional specifically for tests, because it allows you to test what happens when only some of the APIs are called. That being said, I removed these for now since they aren't being used.